subsys/net/lib/aws_jobs/src/aws_jobs.c

```diff
@@ -0,0 +1,397 @@
+
+	struct mqtt_topic subscribe_topic;
+	int err = construct_notify_topic(client->client_id.utf8,
+					 &subscribe_topic);
+	if (err) {
+		return err;
+	}
+
+	const struct mqtt_subscription_list subscription_list = {
+		.list = (struct mqtt_topic *)&subscribe_topic,
+		.list_count = 1,
+		.message_id = SUBSCRIBE_NOTIFY
+	};
+	LOG_DBG("Subscribe: %s", subscribe_topic.topic.utf8);
+
+	return mqtt_subscribe(client, &subscription_list);
+}
+
+
+int aws_jobs_subscribe_expected_topics(struct mqtt_client *const client,
+				       bool notify_next)
+{
+	__ASSERT(client != NULL, "invalid client");
+
+	struct mqtt_topic subscribe_topics[2];
+	char *client_id = client->client_id.utf8;
+
+	int err = construct_get_topic(client_id, &subscribe_topics[0]);
+
+	if (err) {
+		return err;
+	}
+
+	if (notify_next) {
+		err = construct_notify_next_topic(client_id,
+						  &subscribe_topics[1]);
+	} else {
+		err = construct_notify_topic(client_id, &subscribe_topics[1]);
+	}
+
+	if (err) {
+		return err;
+	}
+
+	LOG_DBG("Subscribe: %s", subscribe_topics[0].topic.utf8);
+	LOG_DBG("Subscribe: %s", subscribe_topics[1].topic.utf8);
+
+	const struct mqtt_subscription_list subscription_list = {
+		.list = (struct mqtt_topic *)&subscribe_topics,
+		.list_count = ARRAY_SIZE(subscribe_topics),
+		.message_id = SUBSCRIBE_EXPECTED
+	};
+
+	return mqtt_subscribe(client, &subscription_list);
+}
+
+
+/**
+ * @brief Constructs the get topic for accepted and rejeceted AWS jobs.
+ *
+ * @param[in] client_id Client ID of the current connected MQTT session
+ *	formatted.
+ * @param[in] job_id Job ID of the current accepted job ormatted.
+ * @param[out] topic Outputs the constructed mqtt_topic into the given buffer.
+ *
+ * @retval 0 If sucessful otherwise the return value of report_snprintf_err
+ *
+ */
+static int construct_job_id_get_topic(const u8_t *client_id,
+				      const u8_t *job_id,
+				      struct mqtt_topic *topic)
+{
+	u32_t job_id_get_topic_len;
+	u8_t job_id_get_topic[JOB_ID_GET_TOPIC_MAX_LEN  + 1];
+	int ret = snprintf(job_id_get_topic,
+			  JOB_ID_GET_TOPIC_MAX_LEN,
+			  JOB_ID_GET_TOPIC_TEMPLATE,
+			  client_id,
+			  job_id,
+			  "#");
+	job_id_get_topic_len = ret + 1;
+	ret = report_snprintf_err(ret,
+				  JOB_ID_GET_TOPIC_MAX_LEN,
+				  "job_id_get_topic");
+	if (ret) {
+		return ret;
+	}
+
+	topic->topic.utf8 = job_id_get_topic;
+	topic->topic.size = job_id_get_topic_len;
+	topic->qos = MQTT_QOS_1_AT_LEAST_ONCE;
+	return 0;
+}
+
+
+int subscribe_job_id_get_topic(struct mqtt_client *const client,
+			       const u8_t *job_id)
+{
+	__ASSERT(client != NULL, "invalid client");
+	__ASSERT(job_id != NULL, "invalid job_id");
+
+	struct mqtt_topic subscribe_topic;
+	int err = construct_job_id_get_topic(client->client_id.utf8,
+					     job_id,
+					     &subscribe_topic);
+	if (err) {
+		return err;
+	}
+
+	const struct mqtt_subscription_list subscription_list = {
+		.list = (struct mqtt_topic *)&subscribe_topic,
```

Comment:
Do you need these casts?
Isn't `list` a `struct mqtt_topic *` just like `&subscribe_topic`?

Reply:
You are correct this one does not need a cast